File: app/src/main/java/com/an/github/data/local/dao/GithubDao.java
package com.an.github.data.local.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.an.github.data.local.entity.GithubEntity;

import java.util.List;

@Dao
public interface GithubDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long[] insertRepositories(List<GithubEntity> githubEntities);

    @Query("SELECT * FROM `GithubEntity` where page = :page")
    List<GithubEntity> getRepositoriesByPage(Long page);
}

File: app/src/main/java/com/an/github/data/local/dao/GithubDao.kt
package com.an.github.data.local.dao

import androidx.paging.PagingSource
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.an.github.data.local.entity.GithubEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface GithubDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertRepositories(repositories: List<GithubEntity>)

    @Query("SELECT * FROM GithubEntity")
    fun pagingSource(): PagingSource<Int, GithubEntity>

    @Query("SELECT * FROM GithubEntity")
    fun getRepositories(): List<GithubEntity>

    @Query("SELECT * FROM GithubEntity WHERE remoteId =:remoteId")
    fun getRepository(remoteId: Long): Flow<GithubEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun addRepository(repository: GithubEntity)

    @Query("DELETE FROM GithubEntity")
    suspend fun clearAll()
}
